clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -9028,6 +9028,33 @@ void MemVarRefMigrationRule::runRule(const MatchFinder::MatchResult &Result) {
     };
     const auto *Parent = getParentStmt(MemVarRef);
     bool HasTypeCasted = false;
+    auto Set = DpctGlobalInfo::getUseDeviceGlobalVarSet();
+    auto Info = Global.findMemVarInfo(Decl);
+    std::string Key = Info->getFilePath().getCanonicalPath().str() +
+                      std::to_string(Info->getOffset()) + Info->getName();
+    if (Set.find(Key) == Set.end()) {
```

Comment:
How about only try to find the var in `cudaMemcpy*Symbol` APIs and set the flag?